parsel/unified.py

```diff
@@ -32,22 +28,16 @@ def __init__(self, *args, **kwargs):
 }
 
 
-def _st(response, st):
+def _st(st):
     if st is None:
-        return 'xml' if isinstance(response, XmlResponse) else 'html'
-    elif st in ('xml', 'html'):
+        return 'html'
+    elif st in _ctgroup:
         return st
     else:
         raise ValueError('Invalid type: %s' % st)
 
 
-def _response_from_text(text, st):
-    rt = XmlResponse if st == 'xml' else HtmlResponse
-    return rt(url='about:blank', encoding='utf-8',
-              body=unicode_to_str(text, 'utf-8'))
-
-
-class Selector(object_ref):
+class Selector(object):
 
     __slots__ = ['response', 'text', 'namespaces', 'type', '_expr', '_root',
                  '__weakref__', '_parser', '_csstranslator', '_tostring_method']
```

Comment:
I think it's only `response` that has got to go, right?
`__weakref__` is needed to support weak references -- we want that, right?